### ISSUE TITLE
[BUG] Support series groupby aggregations

### ISSUE THREAD

**Issue by mrocklin** [rapidsai/cudf#535]
**Describe the bug**
Series groupbys don't seem to exist

**Steps/Code to reproduce bug**

```python
import pandas as pd, cudf
import numpy as np
pdf = pd.DataFrame({'x': [1, 2, 3, 4], 'y': [0, 1, 0, 1]})
gdf = cudf.DataFrame.from_pandas(pdf)
gdf.x.groupby(gdf.y + 1).mean()
```

**Comment by thomcom** [comment]
Given that we have `Dataframe` level groupby, adding `Series` groupby should be possible entirely in python.

**Comment by kkraus14** [comment]
Fixed by #895
